### DISCUSSION TITLE
How does vulnerability alerts work?

### DISCUSSION THREAD

**Discussion by gaeljw** [renovatebot/renovate#20697] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.153.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

N/A

### What you tried so far.

I stumbled upon the preset `enableVulnerabilityAlertsWithLabel` and decided to give it a try but so far it didn't raise any PR with the label (totally possible in my repos as we don't have much opened Renovate PR anyway).

Thus I'm wondering how does this feature work:
- does it work with all datasources?
- what's the datasource of vulnerabilities?
- does it work in self-hosted?

I couldn't find anything in the documentation but I may have missed it.

**Comment by HonkingGoose** [comment, marked as answer]
The `:enableVulnerabilityAlertsWithLabel` preset enables the `vulnerabilityAlerts` config option.

Read the [docs for the `vulnerabilityAlerts` config option](https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts) to learn more. Here's the important bit:

> Renovate can read GitHub's Vulnerability Alerts to customize its Pull Requests. For this to work, you must enable the [Dependency graph](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph#enabling-the-dependency-graph), and [Dependabot alerts](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-security-and-analysis-settings-for-your-repository).

**Reply by rarkins** [reply]
If on gitlab, he needs the new experimental vulnerability alerts

**Reply by HonkingGoose** [reply]
Whoops, I missed the part about them running on Git**Lab**. 🙈

**Reply by gaeljw** [reply]
Is there any doc or way to test the experimental stuff for Gitlab?

**Reply by Churro** [reply]
Yes, you can find it here: https://docs.renovatebot.com/configuration-options/#osvvulnerabilityalerts
